### ISSUE TITLE
Keep free positioned node on original location during conversion

### ISSUE THREAD

**Issue by vascotenner** [freeplane/freeplane#361]
### Context
Currently, I am creating a work breakdown for my team. I am using freeplane to organize the required actions. I now want to order these actions in time by moving them to distinct horizontal positions. 

### Goal:
![image](https://user-images.githubusercontent.com/3776190/167782685-acd91d93-286b-4b23-8998-173a70c402d4.png)

In order to do so, I need free positioned nodes, else the two nodes in the example will push eachother away in the vertical direction.

### Start situation:
![image](https://user-images.githubusercontent.com/3776190/167782568-b5f5b4ce-f4d9-41b2-bbd2-cb18159304e3.png)

### Issue:
When converting a node to free positioned node, it position changes to the position of its parent. This is frustrating, as I often already have encoded some information in the position (for example timing in a planning). In my current planning I have 100+ nodes that should be converted.

![image](https://user-images.githubusercontent.com/3776190/167782626-67787e68-97d3-4047-a9fb-fa9463d34627.png)

### Proposed solution
Keep nodes on same location with respect to parent when converting to free positioned node.

**Comment by Danselune** [comment]
Hello, totally agree with the proposed solution, especially in the case you select several nodes from a map and make them free positioned, they all jump on the center of the parents nodes (ones on others). 
That make the job to reposition them quite difficult.

**Comment by macmarrum** [comment]
I would benefit from this feature request too

**Comment by dpolivaev** [comment]
Congratulations. You managed to deliver a clear explanation of your use case and it allowed me to find a solution.
Keeping free positioned nodes would be an extremely difficult task because of the need to handle summary nodes, and outline views, and positions of other nodes.

Instead, I introduced a new layout mode called compact layout. Currently (Freeplane 1.10.3_01), it can be activated at View->View settings->Allow compact layout. It is a property saved with the map and it affects the whole map. If it is chosen, users can move nodes up so that they overlap their upper sibling nodes. I think that it removes all significant limitations on the layout, although the default positions of all nodes remain unchanged.

I am not sure if this property should be defined at the map level or at the node level as it has its pros and cons. In this version it is defined for the whole map, but if you prefer I could define it at the node level like the for the free positioned nodes.

Please test it and give me your feedback (I am going to upload the preview version soon).

**Comment by vascotenner** [comment]
Thank you for providing a solution so quick! From your explanation, this would suffice for me. I am looking forward to test it. Would it be possible to have a portable version on monday morning (tomorrow) 9h CEST?


Dimitry Polivaev ***@***.***> schreef op 15 mei 2022 12:05:46 UTC:

**Comment by dpolivaev** [comment]
https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

**Comment by vascotenner** [comment]
Moving the nodes around works as expect, as long they siblings. In my case, I have sorted my actions in several topics (different branches), but they have dependencies and priorities that are not following the branches:

I like to do this:
![image](https://user-images.githubusercontent.com/3776190/168737083-dc80c9f4-fd56-45ae-9794-7536a35f6b26.png)

But when I try, the top branch gets pushed up instead:
![image](https://user-images.githubusercontent.com/3776190/168737006-85fc66a0-bda6-4b7f-ba84-b2d56bffaa85.png)

**Comment by dpolivaev** [comment]
I fixed it in preview 1.10.3_02

**Comment by vascotenner** [comment]
This is already an improvement, it does not get pushed upward anymore. However, it still pushes parent nodes (including all childeren) downward:
![image](https://user-images.githubusercontent.com/3776190/169564234-d1217f2f-f330-4ec0-9720-dad064c50d62.png)

To get here: 
![image](https://user-images.githubusercontent.com/3776190/169564551-7a830fda-dc1b-4ec8-8f3d-c83070880e37.png)

First move B downward, then move C upward. Doing this in reverse order does not work yet.

**Comment by dpolivaev** [comment]
Yes, it is by intention because moving one node moves all siblings.

**Comment by vascotenner** [comment]
But why are uncle nodes pushed downwards in the first place?

Dimitry Polivaev ***@***.***> schreef op 20 mei 2022 15:48:52 UTC:

**Comment by dpolivaev** [comment]
Sorry, I do not understand the question.

**Comment by dpolivaev** [comment]
> This is already an improvement, it does not get pushed upward anymore. However,

Honestly, I want to be given more and better appreciation for my efforts than that.

**Comment by vascotenner** [comment]
Dimitry, sorry for my very short feedback. I am really impressed by the speed you provided an initial solution and improved it even further. This really helps to work with a 100+ nodes mindmap: One can now place sibling nodes close to eachother. After the first iteration 1.10.3_2, can also move younger nephews (children of the parent node below a node) upwards close to a node. What I find after trying this version, is that one cannot place older nephews downwards to get them close to a node. That is depicted in the first picture in #issuecomment-1133056782, when moving node B downwards, (the parent of) node C gets pushed downwards.
